Dockerfile

```diff
@@ -1,11 +1,14 @@
 FROM alpine
 
 # This installs also pip3
-RUN apk --no-cache add python3 openssh-client git
+RUN apk update \
+    && apk --no-cache add \
+    python3 openssh-client git curl jq \
+    cmd:pip3
 
 # Add the SSH public keys of git hosting providers
 RUN mkdir /root/.ssh
-COPY docker/ssh_known_hosts.txt /root/.ssh/known_hosts
+RUN curl -L https://api.github.com/meta | jq -r '.ssh_keys | .[]' | sed -e 's/^/github.com /' >> /root/.ssh/known_hosts
```

Comment:
We also need to tell curl to fail if the HTTP request is not 200. By default it happily returns 0.
`--fail-with-body`. While at it, you can add `--no-progress-meter`.

Also if we do like this, the pipe can still fail... I have the impression that we are adding new failure modes. If we want to do in this direction, I think we need maybe to use a script instead of a one-liner...

I understand the intent to be DRY and up to date, we even spoke about it :-), but here we are in automation and we have to be careful...

Taking  a step back, GitHub doesn't change its key often. Also if it where, in this dockerfile we would still have to manually build a new image, we do not poll that GitHUB endpoint to see if a new key appeared.

As a first and robust fix, I would just update the existing file. If you then want to explore using the endpoint in a robust way, I am fine if you want to propose another PR.

Opinions?

Reply:
Understand, thanks for the explanation.